C/common/asset_tracking.cpp

```diff
@@ -89,22 +80,8 @@ AssetTracker::AssetTracker(ManagementClient *mgtClient, string service)
 AssetTracker::~AssetTracker()
 {
 	m_shutdown = true;
-	// Signal the worker thread to flush the queue
-	{
-		unique_lock<mutex> lck(m_mutex);
-		m_cv.notify_all();
-	}
-	while (m_pending.size())
-	{
-		// Wait for pending queue to drain
-		this_thread::sleep_for(chrono::milliseconds(10));
-	}
-	if (m_thread)
-	{
-		m_thread->join();
-		delete m_thread;
-		m_thread = NULL;
-	}
+
+	HouseKeeper::getInstance()->addTask(this);
```

Comment:
I don't understand what this is doing. It looks like _addTask(this)_ is adding this instance of AssetTracker to a task list but this AssetTracker instance is about to vanish. The _this_ pointer will become invalid.

Reply:
It should have been removeTask() rather than addTask()